### PR TITLE
Add dataset functions

### PR DESCRIPTION
**Pull request by michaelliangau**
- Delete unused dataset functions
- Consolidate everything into py file

### CODE REVIEW

File: dataset/create_dataset.py [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+    filtered_df[f'f107_index_{i}'] = f107_values[:, i]
+
+if np.isnan(f107_values).any():
+    print("Warning: Some f107 index values are missing.") 
+    import IPython; IPython.embed()
+
+
+# Replace all invalid values with 0
+invalid_values = [999.9, 9.999, 9999.0, 9999.99, 99999.99, 9999999, 9999999.0]
+filtered_df = filtered_df.replace(invalid_values, 0)
```
Comment (blakedehaas) [inline dataset/create_dataset.py:371]: This is not ideal for measurements that do not have 0 in their normal range, as we will be shifting the distribution with an artificial "outlier". I wonder if using the mean of each measurement would be better?

File: dataset/create_dataset.py [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+
+# Save dataset
+train_val, test = train_test_split(filtered_df, test_size=100000, random_state=42)
+train, val = train_test_split(train_val, test_size=100000, random_state=42)
```
Comment (blakedehaas) [inline dataset/create_dataset.py:375]: Random train-test split means we will be testing on points where we have past data and future data. This has some data leakage because of our Chu features that give us the context over a time range, so we are leaking some information about our test set into the Chu features.

**Review by blakedehaas — APPROVED** [review]
Two suggestions:
- Re-evaluate the train-test split method to prevent data leakage via the Chu temporal context features. We want to make sure we are not training on our test set within those Chu temporal features
- Set invalid values for each measurement to the mean of each measurement so we will be centering errors at 0 after standardizing and won't be shifting the distribution when normalizing
